File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int cnt = 0;
        int low = 0;
        int high = people.length - 1;
        
        Arrays.sort(people);
        
        while(low <= high) {
            if (people[low] + people[high] <= limit) {
                low++;
            }
            high--;
            cnt++;
        }
        
        return cnt;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    let count = 0;
    let low = 0;
    let high = people.length - 1;
    
    people.sort((a, b) => a - b);
    
    while(low <= high) {
        if (people[low] + people[high] <= limit) {
            low++;
        }
        high--;
        count++;
    }
    
    return count;
}